src/app/components/PageLayout/ScrollingCard.tsx

```diff
@@ -0,0 +1,19 @@
+import Card from '@mui/material/Card'
+import { FC } from 'react'
+import { useScrolling } from '../../hooks/useScrolling'
+
+/**
+ * This is a wrapper around MUI card that take care of (fixes) automatic scrolling.
+ *
+ * If a Card has an "id" prop, it will add an HTML anchor to the DIV.
+ * It can be added to the URL, and we can scroll to this div.
+ *
+ * However, when loading the page initially, sometimes the HTML element for the card is not yet available in the DOM
+ * when the scrolling is supposed to happen, so it fails. This wrapper component can fix that.
+ *
+ * When the component is mounted, it will check (with reach router) if we are supposed to be scrolled to this element,
+ * and executes the scrolling when necessary.
+ */
+export const ScrollingCard: FC<Parameters<typeof Card>[0]> = props => (
```

Comment:
Is `Parameters<typeof Card>[0]` the same as `CardProps`?

Reply:
id prop is not required

Reply:
> id prop is not required

Yes. And the `useScrolling` hook can handle cases when the input parameter (`targetHash`) is undefined. In that case, it won't do anything.

Reply:
Alright, tho that feels like a bug to me